Add areaOfInterest and detectObject to CV API

mirzamo opened the pull request:
This PR has been approved in the private repo. As we are about to release these features to public, I'm sending this PR.
https://github.com/Azure/azure-rest-api-specs-pr/pull/607

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

mirzamo committed:
Add areaOfInterest and detectObject

* Replace the current swagger with an auto-generated one.
* Add example files for areaOfInterest and detectObject.
* Fix example validation errors for analyze and describe.
* A few fixes for generating java sdk

AutorestCI commented:
# Automation for azure-sdk-for-js

yangyuan commented:
Hi @mirzamo, earlier today I had a meeting with Hui and Cha today.
We will need API Review Board to review the swaggers before we proceed.

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/ComputerVision.json:
```diff
@@ -5,64 +5,15 @@
     "title": "Computer Vision Client",
     "description": "The Computer Vision API provides state-of-the-art algorithms to process images and return information. For example, it can be used to determine if an image contains mature content, or it can be used to find all the faces in an image.  It also has other features like estimating dominant and accent colors, categorizing the content of images, and describing an image with complete ...
   },
-  "securityDefinitions": {
```
securityDefinitions is, in general, before paths. Can you move it back?

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/ComputerVision.json:
```diff
@@ -5,64 +5,15 @@
-      "in": "header"
-    }
-  },
-  "security": [
```
same for "security"

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/ComputerVision.json:
```diff
@@ -5,64 +5,15 @@
-      "apim_key": []
-    }
-  ],
-  "x-ms-parameterized-host": {
```
"x-ms-parameterized-host" is Microsoft's extension for replacing "host", "basePath", "schemes". I think it's better to keep them close to "host"

jianghaolu commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/Ocr.json:
```diff
@@ -227,27 +227,33 @@
+            "InvalidModel",
+            "CancelledRequest",
+            "NotSupportedVisualFeature",
+            "FailedToProcess",
```
This seems to be a breaking change for your API - current SDKs will not be able to handle these new enums. Only enums with "modelAsString" set to true are allowed to add new enum values.

cthrash replied:
Can you help clarify?  If the service returns one of these new values in the JSON, aren't current clients screwed anyway?

mirzamo replied:
@jianghaolu Any comments?

jianghaolu replied:
Yes That's what I mean by saying the rest API is going to break the clients.

jianghaolu replied:
> aren't current clients screwed anyway

This doesn't justify breaking the clients after breaking the APIs. If this list is supposed to grow you should mark "modelAsString" to true in this PR. This PR is bringing breaking changes anyways (unfortunately) but there will be no more in the future.

mirzamo replied:
I changed it to true to prevent future breaking changes. Also [in here](https://github.com/Azure/azure-rest-api-specs/pull/4771/files#diff-adbdfc1efc9df7ca3f9f684ead4e71a5R1183).

mirzamo committed:
Addressing commets: reorder some keys

mirzamo commented:
@yangyuan I addressed all of your comments.
Regarding the API review board, I sent over the swagger and sdk, but the comments are either related to autorest which we don't have any control over it, or naming. We cannot make any changes to the names at this time since the SDKs are already shipped.

yangyuan commented:
@mirzamo 
Can you cc me the email thread? (alias yuanyang)
For existing issues (not related to this PR) we can totally skip them and fix them in next version, but need PM's awareness.

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/examples/SuccessfulDetectObjectsWithStream.json:
```diff
@@ -0,0 +1,35 @@
+        "metadata": {
+          "width": 100,
+          "height": 100,
+          "format": "Jpeg"
```
Just an idea, wouldn't `jpeg` or `JPEG` be better?

mirzamo replied:
I agree, but for all the of other services it's `Jpeg` and I think it's better to keep them consistent.

yangyuan replied:
Just curious, what services you are referring to? I searched before and didn't find anyother..

mirzamo replied:
[Analyze](https://github.com/Azure/azure-rest-api-specs/blob/78e8a4782397b62033139381169eb45bd87e1743/specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/examples/SuccessfulAnalyzeImageByDomainWithStream.json#L15)
[Tag](https://github.com/Azure/azure-rest-api-specs/blob/78e8a4782397b62033139381169eb45bd87e1743/specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/examples/SuccessfulTagImageWithUrl.json#L48)
[Describe](https://github.com/Azure/azure-rest-api-specs/blob/78e8a4782397b62033139381169eb45bd87e1743/specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/examples/SuccessfulDescribeImageWithUrl.json#L39)

yangyuan replied:
Oh you mean other `operations`, I thought you mean other `services`

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/ComputerVision.json:
```diff
@@ -103,7 +76,7 @@
         "responses": {
           "200": {
-            "description": "The response include the extracted features in JSON format.Here is the definitions for enumeration typesClipartTypeNon-clipart = 0,  ambiguous = 1, normal-clipart = 2, good-clipart = 3.LineDrawingTypeNon-LineDrawing = 0,LineDrawing = 1.",
+            "description": "The response include the extracted features in JSON format.Here is the definitions for enumeration typesClipartTypeNon - clipart = 0, ambiguous = 1, normal - clipart = 2, good - clipart = 3.LineDrawingTypeNon - LineDrawing = 0, LineDrawing = 1.",
```
Missed a space or line break?`format.Here`

yangyuan commented on specification/cognitiveservices/data-plane/ComputerVision/stable/v2.0/ComputerVision.json:
```diff
@@ -1309,80 +1324,438 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "The response include the extracted features in JSON format.Here is the definitions for enumeration typesClipartTypeNon - clipart = 0, ambiguous = 1, normal - clipart = 2, good - clipart = 3.LineDrawingTypeNon - LineDrawing = 0, LineDrawing = 1.",
```
Same

yangyuan reviewed (approved):
Look good from my side.
Also passes typo check.

mirzamo committed:
fix formatting

mirzamo committed:
Changing modelAsString parameter to ture to be able to add new values ComputerVisonErrors enum
